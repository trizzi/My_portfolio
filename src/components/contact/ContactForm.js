import React, { useState } from 'react';
import Button from '../button/Button';

const ContactForm = () => {
	const [formData, setFormData] = useState({
		firstName: '',
		lastName: '',
		email: '',
		message: '',
	});

	const handleChange = (e) => {
		const { name, value } = e.target;
		setFormData((prevFormData) => ({
			...prevFormData,
			[name]: value,
		}));
	};

	const handleSubmit = async (e) => {
		e.preventDefault();

		try {
			const response = await fetch('/.netlify/functions/submit-form', {
				method: 'POST',
				headers: {
					'Content-Type': 'application/json',
				},
				body: JSON.stringify(formData),
			});

			if (response.ok) {
				// Form submission successful
				console.log('Form submitted successfully');
			} else {
				// Form submission failed
				console.error('Form submission failed');
			}
		} catch (error) {
			console.error('An error occurred while submitting the form:', error);
		}
	};

	return (
		<form
			className='flex flex-col items-center justify-center bg-black py-8'
			onSubmit={handleSubmit}
			data-netlify='true'>
			<input
				type='hidden'
				name='form-name'
				value='contact v1'
			/>
			<div hidden>
				<label>
					Don’t fill this out if you’re human: <input name='bot-field' />
				</label>
			</div>

			<div className='flex justify-between gap-4'>
				<label htmlFor='First Name'>
					First Name
					<input
						type='text'
						id='first_name'
						name='firstName'
						className='mb-5 block w-40 text-black rounded-md border-0 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
						value={formData.firstName}
						onChange={handleChange}
						placeholder='First Name'
					/>
				</label>
				<label htmlFor='Last Name'>
					Last Name
					<input
						type='text'
						id='last_name'
						name='lastName'
						className='mb-5 block w-40 text-black rounded-md border-0 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
						value={formData.lastName}
						onChange={handleChange}
						placeholder='Last Name'
					/>
				</label>
			</div>

			<label htmlFor='Email'>
				Email
				<input
					type='text'
					id='email'
					name='email'
					className='mb-5 block w-[21rem] text-black rounded-md border-0 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
					value={formData.email}
					onChange={handleChange}
					placeholder='Email'
				/>
			</label>
			<label htmlFor='Message'>
				Message
				<textarea
					type='text'
					id='message'
					name='message'
					className='mb-5 block w-[21rem] text-black rounded-md border-0 placeholder:text-gray-400 focus:ring-2 focus:ring-inset focus:ring-indigo-600 sm:text-sm sm:leading-6'
					value={formData.message}
					onChange={handleChange}
					placeholder='Type Message ...'
				/>
			</label>

			<Button
				text={'Send'}
				background
				type='submit'
			/>
		</form>
	);
};

export default ContactForm;
