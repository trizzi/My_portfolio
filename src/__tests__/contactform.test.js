import React from 'react';
import { render, fireEvent, waitFor, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import { ThemeContextProvider } from '../context/ThemeContext';

import ContactForm from '../components/contact/ContactForm';

test('submits the form with user input', async () => {
	// Mock fetch function to simulate successful form submission
	jest.spyOn(global, 'fetch').mockResolvedValue({ ok: true });

	// Mock console.log to capture the message
	const consoleSpy = jest.spyOn(console, 'log').mockImplementation();

	// Render the ContactForm component
	const { getByLabelText, getByText } = render(
		<ThemeContextProvider>
			<ContactForm />
		</ThemeContextProvider>,
	);

	// Fill out the form inputs
	fireEvent.change(screen.getByPlaceholderText('First Name'), {
		target: { value: 'John' },
	});
	fireEvent.change(screen.getByPlaceholderText('Last Name'), {
		target: { value: 'Doe' },
	});
	fireEvent.change(screen.getByPlaceholderText('Email'), {
		target: { value: 'john@example.com' },
	});
	fireEvent.change(screen.getByPlaceholderText('Type Message ...'), {
		target: { value: 'Hello, this is a test message.' },
	});

	// Submit the form
	fireEvent.click(screen.getByText('Send'));

	// Wait for form submission
	await waitFor(() => {
		// Assert that form submission is successful
		expect(consoleSpy).toHaveBeenCalledWith('Form submitted successfully');
	});

	// Clean up the mock
	global.fetch.mockRestore();

	expect(getByLabelText).toMatchSnapshot();
});
