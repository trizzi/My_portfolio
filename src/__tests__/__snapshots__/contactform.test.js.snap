// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`submits the form with user input 1`] = `[Function]`;
